.github/workflows/main.yml

```diff
@@ -19,8 +19,8 @@ jobs:
         - { name: Windows (mingw64+CMake),      os: windows-latest, shell: 'msys2 {0}', msystem: mingw64, msys-env: mingw-w64-x86_64,
             cmake: '-DSDL2IMAGE_BACKEND_STB=OFF -DSDL2IMAGE_BACKEND_WIC=OFF -DSDL2IMAGE_VENDORED=OFF -DSDL2IMAGE_AVIF=ON -G "Ninja Multi-Config"' }
         - { name: Linux (autotools),            os: ubuntu-20.04,   shell: sh }
-        - { name: Linux (CMake),                os: ubuntu-20.04,   shell: sh,   cmake: '-DSDL2IMAGE_VENDORED=ON -GNinja' }
-        - { name: 'Linux (CMake, static)',      os: ubuntu-20.04,   shell: sh,   cmake: '-DSDL2IMAGE_VENDORED=ON -DBUILD_SHARED_LIBS=OFF -GNinja' }
+        - { name: Linux (CMake),                os: ubuntu-20.04,   shell: sh,   cmake: '-DSDL2IMAGE_TESTS=ON -DSDL2IMAGE_VENDORED=ON -GNinja ' }
+        - { name: 'Linux (CMake, static)',      os: ubuntu-20.04,   shell: sh,   cmake: '-DSDL2IMAGE_TESTS=ON -DSDL2IMAGE_VENDORED=ON -DBUILD_SHARED_LIBS=OFF -GNinja' }
```

Comment:
Can we move `-DSDL2IMAGE_TESTS=ON` to the workflow steps?
Are did you put them here because of the tests failing on Macos?

Reply:
I put them here because there were enough exceptions and maybes that this seemed clearer than the way I'd previously been doing it, particularly while I was juggling all the various builds to get as many tests passing as I could.

Now that I've sorted out what does and doesn't work, they could move back to the `cmake` invocation as "only if Linux", I think.

Reply:
> did you put them here because of the tests failing on Macos?

No, they're here because the tests don't compile on macOS with CMake (it looks like Autotools can find `-lSDL_test`, but CMake can't, for whatever reason).

Reply:
I think it's because we want to link to `SDL2_test` as `-lSDL2_test`, but with Brew libraries are not available in the usual locations.
Perhaps we need to do things like:
```cmake
find_library(SDL2_TEST_LIBRARY
     NAMES SDL2_test
)
if(SDL2_TEST_LIBRARY AND NOT TARGET SDL2::SDL2test)
    add_library(SDL2::SDL2test UNKNOWN IMPORTED)
    set_target_properties(SDL2::SDL2test PROPERTIES
        IMPORTED_LOCATION "${SDL2_TEST_LIBRARY}"
    )
endif()
```
+ throw an error when SDL2_test is not found.

Reply:
I believe this PR is an improvement with no regressions: it fixes some bugs (not just in the test but also in the functionally necessary code), and the number of platforms tested goes up. Previously we weren't running tests for macOS at all, now we're testing macOS+Autotools which is better than nothing; and previously we were running tests with Linux+Autotools, but now we're also testing Linux+CMake.

Also the branch is getting quite long. So if possible I'd prefer to get the changes-so-far merged before doing the necessary trial-and-error to find this library on a remote system I can only access via CI.

Reply:
I'll try this:

> Can we move -DSDL2IMAGE_TESTS=ON to the workflow steps?

but I'm not intending to include running the test on macOS+CMake in the scope of this branch.

Reply:
It won't build on Macos due to SDL2_test not being found. So I wouldn't bother.

Reply:
SDL_test not found on macOS → #260